### PR TITLE
Reduce restart memory usage during SP

### PR DESCRIPTION
**Pull request by rgknox**
### Description:

Satellite Phenology mode only has one cohort per patch. Restart vectors don't require large allocations due to this simplification.  Modifications are made to prevent large allocations, including the blocking of litter data in restart files.
Answers are not expected to change, in any mode.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/master/CONTRIBUTING.md) document.
- [ ] FATES PASS/FAIL regression tests were run (TBD)
- [x] If answers were expected to change, evaluation was performed and provided

### Test Results:

CTSM (or) E3SM (specify which) test hash-tag:

CTSM (or) E3SM (specify which) baseline hash-tag:

FATES baseline hash-tag:

Test Output:

### CODE REVIEW

File: main/FatesRestartInterfaceMod.F90 [hunk 1]
```diff
@@ -3447,10 +3462,8 @@ subroutine update_3dpatch_radiation(this, nsites, sites, bc_out)
                  bc_out(s)%fabi_parb(ifp,:) = 0.0_r8
                  do ib = 1,hlm_numSWb
 
-                    ! REQUIRES A FIX HERE albd vs albi
-
                     bc_out(s)%albd_parb(ifp,ib) = currentPatch%gnd_alb_dir(ib)
-                    bc_out(s)%albd_parb(ifp,ib) = currentPatch%gnd_alb_dif(ib)
+                    bc_out(s)%albi_parb(ifp,ib) = currentPatch%gnd_alb_dif(ib)
```
Comment (rgknox) [inline main/FatesRestartInterfaceMod.F90:3466]: hmmm, might not be B4B.. should move this to a different PR

**Comment by rosiealice** [comment]
Ah, now I understand. This looks good @rgknox

File: main/FatesRestartInterfaceMod.F90 (hunk 1 above)
Comment (rgknox) [reply main/FatesRestartInterfaceMod.F90]: reverted and moved to other pr

**Review by glemieux — APPROVED** [review]
Just a couple questions regarding the selection of potential maximum values for the `maxElementsperPatch` and `...perSite` updates.

File: main/FatesInterfaceMod.F90 [hunk 2]
```diff
@@ -835,17 +836,14 @@ subroutine SetFatesGlobalElements2(use_fates)
...
-         
-         if (fates_maxPatchesPerSite * fates_maxElementsPerPatch <  numWaterMem) then
-            write(fates_log(), *) 'By using such a tiny number of maximum patches and maximum cohorts'
-            write(fates_log(), *) ' this could create problems for indexing in restart files'
-            write(fates_log(), *) ' The multiple of the two has to be greater than numWaterMem'
-            call endrun(msg=errMsg(sourcefile, __LINE__))
+         if(hlm_use_sp.eq.itrue) then
+            fates_maxElementsPerPatch = maxSWb
+         else
+            fates_maxElementsPerPatch = max(maxSWb,max_cohort_per_patch, ndcmpy*hlm_maxlevsoil ,ncwd*hlm_maxlevsoil)
```
[the remark this answers is not in the document]
Comment (glemieux) [reply main/FatesInterfaceMod.F90:842]: Ryan and I discussed this offline.

**Comment by glemieux** [comment]
Regression testing underway on cheyenne.

**Comment by glemieux** [comment]
Regression testing is resulting in failures for the sp mode test mod runs due to this bounds check in `allocate_bcin`:
https://github.com/NGEET/fates/blob/75372955e27e53a70978afad955811d8db7a03eb/main/FatesInterfaceMod.F90#L408-L409

```
95: The restart files require that space is allocated
95: to accomodate the multi-dimensional patch arrays
95: that are nlevsoil*numpft and nlevsoil*ncwd
95: fates_maxElementsPerPatch =            2
95: nlevsoil =           20
95: dcmpy =            3
95: ncwd  =            4
95: numpft*nlevsoil =          240
95: ncwd*nlevsoil =           80
95: To increase max_elements, change numlevsoil_max
95: ENDRUN:
95: ERROR in FatesInterfaceMod.F90 at line 420     
```
I think we just need to add a check for sp mode here to avoid this specific check.

**Comment by rgknox** [comment]
@glemieux I updated the code to address your comments and the failure

**Comment by glemieux** [comment]
Regression testing on Cheyenne is complete.  All expected tests are b4b against `fates-sci.1.65.3_api.25.4.0-ctsm5.1.dev121`

Folder: `/glade/u/home/glemieux/scratch/ctsm-tests/tests_pr991-allocate-fix`
